ticdc/ticdc-architecture.md

```diff
@@ -0,0 +1,180 @@
+
+### Architecture-related concepts
+
+- Capture: The process that runs the TiCDC node. Multiple Capture processes form a TiCDC cluster. The Capture process is responsible for replicating data changes in TiKV, including receiving and actively pulling data changes, and replicating the data to the downstream.
+- Cpature Owner: The owner Capture among multiple Capture processes. Only one such role exists in a TiCDC cluster at a time. The Capture Owner is responsible for scheduling data within the cluster.
+- Processor: The logical thread inside Capture. Each Processor is responsible for processing the data of one or more tables in the same replication stream. A Capture node can run multiple Processors.
+- Changefeed: A task that replicates data from an upstream TiDB cluster to a downstream system. A changefeed contains multiple Tasks, and each Task is processed by a Capture node.
+
+### Timestamp-related concepts
+
+TiCDC introduces a series of timestamps (TS) to indicate the status of data replication. These timestamps are used to ensure that data is replicated to the downstream at least once and that the consistency of data is guaranteed.
+
+#### ResolvedTS
+
+This timestamp exists in both TiKV and TiCDC.
+
+- ResolvedTS in TiKV: Represents the start time of the earliest transaction in a Region leader, that is, `ResolvedTS` = max(`ResolvedTS`, min(`StartTS`)). Because a TiDB cluster contains multiple TiKV nodes, the minimum ResolvedTS of the Region leader on all TiKV nodes is called the Global ResolvedTS. The TiDB cluster ensures that all transactions before the Global ResolvedTS are committed. Alternatively, you can assume that there are no uncommitted transactions before this timestamp.
+
+- ResolvedTS in TiCDC:
+
+    - table ResolvedTS: Each table has a table-level ResolvedTS, which indicates the minimum data changes to this table received from TiKV. To make it simple, this timestamp is the same as the minimum value of the ResolvedTS of all Regions corresponding to this table on the TiKV node.
+    - global ResolvedTS: The minimum ResolvedTS of all Processors on all TiCDC nodes. Because each TiCDC node has one or more Processors, each Processor corresponds to multiple table pipelines.
+
+    For TiCDC, the ResolvedTS sent by TiKV is a special event in the format of `<resolvedTS: timestamp>`. In general, the ResolvedTS satisfies the following constraints:
+
+    ```
+    table resolved TS >= local resolved TS >= global Resolved TS
+    ```
+
+#### CheckpointTS
+
+This timestamp only exists in TiCDC. It indicates the minimum data changes that have been replicated to the downstream. That is, for TiCDC, the data before this timestamp has been replicated to the downstream system.
+
+- Table checkpointTS: Because TiCDC replicates data in tables, the table checkpointTS indicates the minimum data that has been replicated at the table level.
```

Comment:
```suggestion
- Table checkpointTS: Because TiCDC replicates data in tables, the table checkpointTS indicates the data order than checkpointTS that has been replicated at the table level.
```

Reply:
不够准确，应该是”最大的并且小于它的数据都已经被复制到下游“